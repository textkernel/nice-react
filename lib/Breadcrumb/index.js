export { default as BreadcrumbGroup } from './BreadcrumbGroup';
export { default as BreadcrumbItem } from './BreadcrumbItem';