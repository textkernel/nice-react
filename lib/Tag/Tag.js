import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Item from './Item';
import Label from './Label';
import Button from './Button';
import Badge from './Badge';


class Tag extends Component {

    
    static Label = Label;
    static Item = Item;
    static Button = Button;
    static Badge = Badge;

    
    static propTypes = {
        children: PropTypes.node.isRequired,
        className: PropTypes.string
    }


    static defaultProps = {
        className: ''
    }
    

    render() {
        const { children, className, ...rest } = this.props;

        return (
            <div className={classnames('nice-tags-container', className)} {...rest}>
                { children }
            </div>  
        );
    }

}


export default Tag;