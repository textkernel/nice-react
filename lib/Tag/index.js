export { default as Tags } from './Tags';
export { default as Tag } from './Tag';