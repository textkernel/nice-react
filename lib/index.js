import '@textkernel/nice/dist/nice.min.css';

export { default as Panel } from './Panel';
export { Button, ButtonGroup } from './Buttons';
export { default as Progress } from './Progress';
export { default as Loading } from './Loading';
export { BreadcrumbGroup, BreadcrumbItem } from './Breadcrumb';