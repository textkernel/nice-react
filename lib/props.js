const contexts = ['brand', 'link', 'primary', 'accent', 'good', 'warning', 'bad'];
const sizes = ['small', 'medium', 'large'];
 
const props = {
  contexts,
  buttonContexts: contexts,
  buttonSizes: sizes 
};

module.exports = props;