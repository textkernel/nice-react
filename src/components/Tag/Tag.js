import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Label from './Label';
import Badge from './Badge';
import Button from './Button';
import './Tags.scss';

export default class Tag extends Component {


    static propTypes = {
        children: PropTypes.oneOfType([
            PropTypes.node,
            PropTypes.array,
            PropTypes.instanceOf(Label),
            PropTypes.instanceOf(Badge),
            PropTypes.instanceOf(Button),
        ]).isRequired,
        className: PropTypes.string,
        style: PropTypes.object,
        context: PropTypes.string,
        onClick: PropTypes.func,
    }


    static defaultProps = {
        className: '',
        style: {},
        context: '',
        onClick() {}
    };

    
    static Button = Button;
    static Badge = Badge;
    static Label = Label;
    

    handleClick = (event) => {
        const { onClick } = this.props;
        event.stopPropagation();

        onClick(event);
    }


    render() {
        const { children, style, className, context, onClick, ...rest } = this.props;

        return (
            <div 
                className={classnames('nice-tag', context, className)}
                style={style}
                onClick={this.handleClick}
                role="button"
                tabIndex="-1"
                { ...rest }
            >
                {children}
            </div>
        );
    }
}