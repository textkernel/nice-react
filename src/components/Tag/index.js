export { default } from './Tags';
export { default as Tag } from './Tag';