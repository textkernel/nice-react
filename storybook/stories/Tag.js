import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobsOptions, withKnobs, text, boolean, select, number } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import props from '../../lib/props';

import { Tags, Tag } from 'nice-react';


const stories = storiesOf('Tag', module);


stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);


stories.add('Default', () => {
    return (
        <Tags>
            <Tag>
                <Tag.Label label='Netherlands' />
            </Tag>
        </Tags>
    );
});


stories.add('with input', () => {
    return (
        <Tags>
            <Tag>
                <Tag.Label label='Netherlands' />
            </Tag>
            <Tags.Input placeholder='Add new tag...' />
        </Tags>
    );
});


stories.add('with button', () => {
    return (
        <Tags>
            <Tag>
                <Tag.Label label='Netherlands' />
                <Tag.Button icon='fa-trash-o' />
            </Tag>
        </Tags>
    );
});


stories.add('with badge', () => {
    return (
        <Tags>
            <Tag>
                <Tag.Label label='Netherlands' />
                <Tag.Badge label='8' />
            </Tag>
        </Tags>
    );
});


stories.add('with badge and sub-items', () => {
    return (
        <Tags>
            <Tag>
                <Tag.Label label='Netherlands' />
                <Tag.Badge label='8'>
                    <Tag.Badge.Item>Item 1</Tag.Badge.Item>
                    <Tag.Badge.Item>Item 2</Tag.Badge.Item>
                </Tag.Badge>
            </Tag>
        </Tags>
    );
});


stories.add('with badge/sub-items and button', () => {
    return (
        <Tags>
            <Tag>
                <Tag.Label label='Netherlands' />
                <Tag.Badge label='8'>
                    <Tag.Badge.Item>Item 1</Tag.Badge.Item>
                    <Tag.Badge.Item>Item 2</Tag.Badge.Item>
                </Tag.Badge>
                <Tag.Button icon='fa-trash-o' />
            </Tag>
        </Tags>
    );
});


stories.add('with dynamic properties', () => {
    const contextOptions = props.tagContexts;
    const label = text('Tag Label', 'Netherlands');
    const badgeLabel = text('Badge Label', '10');
    const context = select('Context', contextOptions, contextOptions[0]);
    const showButton = boolean('Show Button', true);
    const showBadge = boolean('Show Badge', true);


    return (
        <Tags>
            <Tag context={context}>
                <Tag.Label label={label} />
                { showBadge && 
                    <Tag.Badge label={badgeLabel}>
                        <Tag.Badge.Item>Item 1</Tag.Badge.Item>
                        <Tag.Badge.Item>Item 2</Tag.Badge.Item>
                    </Tag.Badge>
                }
                { showButton && <Tag.Button icon='fa-trash-o' /> }
            </Tag>
        </Tags>
    );
});