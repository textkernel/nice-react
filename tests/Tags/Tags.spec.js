import React from 'react';
import { Tags, Tag } from '../../lib/Tag';

describe('<Tags /> <Tag />', () => {
    it('should render simple tag', () => {
        const wrapper = shallow(
            <Tags>
                <Tag>
                    <Tag.Label label="Germany" />
                </Tag>
            </Tags>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with badge without items', () => {
        const wrapper = shallow(
            <Tags>
                <Tag>
                    <Tag.Label label="Germany" />
                    <Tag.Badge label="16" />
                </Tag>
            </Tags>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with badge with items', () => {
        const wrapper = shallow(
            <Tags>
                <Tag>
                    <Tag.Label label="Germany" />
                    <Tag.Badge label="16">
                        <Tag.Badge.Item>Item 1</Tag.Badge.Item>
                        <Tag.Badge.Item>Item 2</Tag.Badge.Item>
                    </Tag.Badge>
                </Tag>
            </Tags>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with badge and button', () => {
        const wrapper = shallow(
            <Tags>
                <Tag>
                    <Tag.Label label="Germany" />
                    <Tag.Badge label="16">
                        <Tag.Badge.Item>Item 1</Tag.Badge.Item>
                        <Tag.Badge.Item>Item 2</Tag.Badge.Item>
                    </Tag.Badge>

                    <Tag.Button icon="fa-trash-o" />

                </Tag>
            </Tags>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with the input for new tags', () => {
        const wrapper = shallow(
            <Tags>
                <Tag>
                    <Tag.Label label="Germany" />
                </Tag>
                <Tags.Input placeholder="Add new tag..." />
            </Tags>
        );
        expect(wrapper).toMatchSnapshot();
    });

    
    it('should trigger onClick callback of Button', () => {
        const mockCallback = jest.fn();

        const wrapper = mount(
            <Tags>
                <Tag>
                    <Tag.Button onClick={mockCallback} icon="fa-trash-o" />
                </Tag>
            </Tags>
        );
        wrapper.find('button.nice-tag-btn').simulate('click', { preventDefault() {} });
        expect(mockCallback).toHaveBeenCalled();
    });


    it('should trigger onClick callback when clicks on Tag', () => {
        const mockCallback = jest.fn();

        const wrapper = mount(
            <Tags>
                <Tag onClick={mockCallback}>
                    <Tag.Label label="Germany" />
                </Tag>
            </Tags>
        );
        wrapper.find('div.nice-tag').simulate('click', { preventDefault() {} });
        expect(mockCallback).toHaveBeenCalled();
    });

    it('should trigger onClick callback when clicks on Tag Item', () => {
        const mockCallback = jest.fn();

        const wrapper = mount(
            <Tags>
                <Tag onClick={mockCallback}>
                    <Tag.Label label="Germany" />
                    <Tag.Badge label="16">
                        <Tag.Badge.Item onClick={mockCallback}> Item </Tag.Badge.Item>
                    </Tag.Badge>
                </Tag>
            </Tags>
        );
        wrapper.find('li a').simulate('click', { preventDefault() {} });
        expect(mockCallback).toHaveBeenCalled();
    });

    it('should trigger onClick callback when clicks on Tag Badge', () => {
        const mockCallback = jest.fn();

        const wrapper = mount(
            <Tags>
                <Tag onClick={mockCallback}>
                    <Tag.Label label="Germany" />
                    <Tag.Badge label="16" onClick={mockCallback}>
                        <Tag.Badge.Item> Item </Tag.Badge.Item>
                    </Tag.Badge>
                </Tag>
            </Tags>
        );
        wrapper.find('div.nice-tag-badge').simulate('click', { preventDefault() {} });
        expect(mockCallback).toHaveBeenCalled();
    });

});