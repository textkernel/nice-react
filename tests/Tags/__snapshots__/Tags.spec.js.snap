// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> <Tag /> should render Tags with custom className/styles 1`] = `
<Tags
  className="my-custom-classname"
  style={
    Object {
      "top": 0,
    }
  }
>
  <div
    className="nice-tags-container my-custom-classname"
    style={
      Object {
        "top": 0,
      }
    }
  >
    <Tag
      className="my-custom-classname"
      context=""
      onClick={[Function]}
      style={
        Object {
          "top": 0,
        }
      }
    >
      <div
        className="nice-tag my-custom-classname"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "top": 0,
          }
        }
        tabIndex="-1"
      >
        <Label
          className="my-custom-classname"
          label="Germany"
          style={
            Object {
              "top": 0,
            }
          }
        >
          <div
            className="nice-tag-label my-custom-classname"
            style={
              Object {
                "top": 0,
              }
            }
          >
            Germany
          </div>
        </Label>
        <Badge
          className="my-custom-classname"
          label="16"
          onClick={[Function]}
          style={
            Object {
              "top": 0,
            }
          }
        >
          <div
            className="nice-tag-badge my-custom-classname"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "top": 0,
              }
            }
            tabIndex="-1"
          >
            16
            <div
              className="nice-tag-dropdown hover"
            >
              <ul>
                <Item
                  className="my-custom-classname"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "top": 0,
                    }
                  }
                >
                  <li
                    className="my-custom-classname"
                    style={
                      Object {
                        "top": 0,
                      }
                    }
                  >
                    <a
                      href="#"
                      onClick={[Function]}
                    >
                       Item 
                    </a>
                  </li>
                </Item>
              </ul>
            </div>
          </div>
        </Badge>
      </div>
    </Tag>
    <Input
      className="my-custom-classname"
      onChange={null}
      placeholder="Add new tag..."
      style={
        Object {
          "top": 0,
        }
      }
      value=""
    >
      <input
        className="nice-tag-input my-custom-classname"
        onChange={[Function]}
        placeholder="Add new tag..."
        style={
          Object {
            "top": 0,
          }
        }
        type="text"
        value=""
      />
    </Input>
  </div>
</Tags>
`;

exports[`<Tags /> <Tag /> should render simple tag 1`] = `
<div
  className="nice-tags-container"
  style={Object {}}
>
  <Tag
    className=""
    context=""
    onClick={[Function]}
    style={Object {}}
  >
    <Label
      className=""
      label="Germany"
      style={Object {}}
    />
  </Tag>
</div>
`;

exports[`<Tags /> <Tag /> should render with badge and button 1`] = `
<div
  className="nice-tags-container"
  style={Object {}}
>
  <Tag
    className=""
    context=""
    onClick={[Function]}
    style={Object {}}
  >
    <Label
      className=""
      label="Germany"
      style={Object {}}
    />
    <Badge
      className=""
      label="16"
      onClick={[Function]}
      style={Object {}}
    >
      <Item
        className=""
        href="#"
        onClick={[Function]}
        style={Object {}}
      >
        Item 1
      </Item>
      <Item
        className=""
        href="#"
        onClick={[Function]}
        style={Object {}}
      >
        Item 2
      </Item>
    </Badge>
    <Button
      className=""
      icon="fa-trash-o"
      onClick={[Function]}
      style={Object {}}
    />
  </Tag>
</div>
`;

exports[`<Tags /> <Tag /> should render with badge with items 1`] = `
<div
  className="nice-tags-container"
  style={Object {}}
>
  <Tag
    className=""
    context=""
    onClick={[Function]}
    style={Object {}}
  >
    <Label
      className=""
      label="Germany"
      style={Object {}}
    />
    <Badge
      className=""
      label="16"
      onClick={[Function]}
      style={Object {}}
    >
      <Item
        className=""
        href="#"
        onClick={[Function]}
        style={Object {}}
      >
        Item 1
      </Item>
      <Item
        className=""
        href="#"
        onClick={[Function]}
        style={Object {}}
      >
        Item 2
      </Item>
    </Badge>
  </Tag>
</div>
`;

exports[`<Tags /> <Tag /> should render with badge without items 1`] = `
<div
  className="nice-tags-container"
  style={Object {}}
>
  <Tag
    className=""
    context=""
    onClick={[Function]}
    style={Object {}}
  >
    <Label
      className=""
      label="Germany"
      style={Object {}}
    />
    <Badge
      className=""
      label="16"
      onClick={[Function]}
      style={Object {}}
    />
  </Tag>
</div>
`;

exports[`<Tags /> <Tag /> should render with the input for new tags 1`] = `
<div
  className="nice-tags-container"
  style={Object {}}
>
  <Tag
    className=""
    context=""
    onClick={[Function]}
    style={Object {}}
  >
    <Label
      className=""
      label="Germany"
      style={Object {}}
    />
  </Tag>
  <Input
    className=""
    onChange={null}
    placeholder="Add new tag..."
    style={Object {}}
    value=""
  />
</div>
`;
